src/test/java/org/fogbowcloud/manager/core/processors/FulfilledProcessorTest.java

```diff
@@ -99,355 +95,365 @@ public void tearDown() {
+    // list, and removed from the Fulfilled list.
+    @SuppressWarnings("static-access")
     @Test
-    public void testRunProcessLocalComputeOrderWhenGetInstanceFailed()
+    public void testRunProcessLocalComputeOrderWhenInstanceStateIsFailed()
             throws FogbowManagerException, UnexpectedException, InterruptedException {
-        
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
-
-        Mockito.doReturn(null).when(this.localCloudConnector).getInstance(Mockito.any(Order.class));
-
         Assert.assertNull(this.failedOrderList.getNext());
 
+        // exercise
+        Mockito.doReturn(null).when(this.localCloudConnector).getInstance(Mockito.any(Order.class));
         spyFulfiledProcessor();
 
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
+        this.thread.sleep(DEFAULT_SLEEP_TIME);
 
-        Thread.sleep(MIN_SLEEP_TIME);
-
-        Assert.assertNotNull(this.failedOrderList.getNext());
+        // verify
+        Order test = this.failedOrderList.getNext();
+        Assert.assertNotNull(test);
+        Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
+        Assert.assertEquals(OrderState.FAILED, test.getOrderState());
         Assert.assertNull(this.fulfilledOrderList.getNext());
     }
 
+    // test case: In calling the processFulfilledOrder() method for any order other than Fulfilled,
+    // you must not make state transition by keeping the order in your source list.
     @Test
     public void testProcessComputeOrderNotFulfilled()
             throws FogbowManagerException, UnexpectedException, InterruptedException {
-        
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FAILED);
         this.failedOrderList.addItem(order);
+        Assert.assertNull(this.fulfilledOrderList.getNext());
 
+        // exercise
         spyFulfiledProcessor();
-
         this.fulfilledProcessor.processFulfilledOrder(order);
+
+        // verify
+        Assert.assertNull(this.fulfilledOrderList.getNext());
         Assert.assertNotNull(this.failedOrderList.getNext());
     }
 
+    // test case: When running thread in the FulfilledProcessor without a LocalMember, the method
+    // processFulfilledOrder() must not change OrderState to Failed and must remain in Fulfilled
+    // list.
     @Test
     public void testRunProcessLocalComputeOrderWithoutLocalMember()
             throws FogbowManagerException, UnexpectedException, InterruptedException {
-        
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         this.fulfilledProcessor = Mockito.spy(new FulfilledProcessor(REMOTE_MEMBER_ID,
```

Comment:
Do we really need to spy on this object?

Reply:
No, I will remove the spy from this object.